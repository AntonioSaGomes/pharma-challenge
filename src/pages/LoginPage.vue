<template>
  <div class="login-container-wrapper">
    <div class="login-container">
      <img class="logo-icon" src="/logo-white.svg" />
      <p>Access our platform too see some amazing content</p>
      <CustomButton class="login-btn" text="Login" @click="login" />
    </div>
    <img class="login-img" src="login-image.png" />
  </div>
</template>
<script lang="ts">
import CustomButton from "../material/CustomButton/CustomButton.vue";

export default {
  methods: {
    login() {
      this.$auth0.loginWithRedirect();
    },
  },
  components: {
    CustomButton,
  },
};
</script>

<style scoped>
.social-logins-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.login-container-wrapper {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  place-items: center;
  overflow-y: auto;
  padding: 4rem;
  box-sizing: border-box;
  height: 100%;
  width: 100%;
}
.login-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 3rem;
  align-self: center;
}

.title {
  text-align: left;
}

.logo-icon {
  width: 10rem;
  height: 6rem;
}

.login-img {
  max-width: 100%;
  height: auto;
  width: auto\9;
}

@media screen and (max-width: 800px) {
  .login-container-wrapper {
    padding: 1rem;
  }
}
@media screen and (max-width: 600px) {
  .login-container-wrapper {
    grid-template-columns: 1fr;
    padding: 0rem;
  }
}
</style>
