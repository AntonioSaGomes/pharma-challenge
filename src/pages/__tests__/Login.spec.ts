import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import LoginPage from "../LoginPage.vue";

describe("LoginPage", () => {
  it("renders the component correctly", () => {
    const wrapper = mount(LoginPage);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("displays the login button", () => {
    const wrapper = mount(LoginPage);
    const loginButton = wrapper.find("button");
    expect(loginButton.text()).toBe("Login");
  });
});
