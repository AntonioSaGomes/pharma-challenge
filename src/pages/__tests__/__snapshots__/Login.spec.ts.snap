// Vitest Snapshot v1

exports[`LoginPage > renders the component correctly 1`] = `
"<div class=\\"login-container-wrapper\\" data-v-eec819be=\\"\\">
  <div class=\\"login-container\\" data-v-eec819be=\\"\\"><img class=\\"logo-icon\\" src=\\"/logo-white.svg\\" data-v-eec819be=\\"\\">
    <p data-v-eec819be=\\"\\">Access our platform too see some amazing content</p><button style=\\"opacity: 0.9;\\" class=\\"login-btn\\" data-v-14b65c88=\\"\\" data-v-eec819be=\\"\\">Login</button>
  </div><img class=\\"login-img\\" src=\\"login-image.png\\" data-v-eec819be=\\"\\">
</div>"
`;
